site/src/pages/RoadmapPage/sidebar/MajorRequiredCourseList.tsx

```diff
@@ -182,46 +185,84 @@ const MajorRequiredCourseList: FC = () => {
-          isDisabled={specsLoading}
-          isLoading={specsLoading}
-          onChange={(data) => {
-            if (data!.value.id === selectedSpec?.id) return;
-            setResultsLoading(true);
-            dispatch(setRequirements([])); // set to empty immediately because otherwise it's out of date
-            dispatch(setSpecialization(data!.value));
-            saveSpecialization(data!.value);
-          }}
-          className="ppc-combobox"
-          classNamePrefix="ppc-combobox"
-          placeholder="Select a specialization..."
-          theme={(t) => comboboxTheme(t, isDark)}
-        />
-      )}
-      {selectedMajor && (!selectedMajor.specializations.length || selectedSpec) && (
-        <>{resultsLoading ? loadingIcon : <ProgramRequirementsList requirements={requirements} />}</>
-      )}
+
+      {selectedMajors.map((majorWithSpec) => (
+        <div key={majorWithSpec.major.id} className="major-section mt-3">
+          <h4>{majorWithSpec.major.name}</h4>
+          {majorWithSpec.major.specializations.length > 0 && (
+            <Select
+              options={specOptions[majorWithSpec.major.id] || []}
+              value={
+                majorWithSpec.specialization
+                  ? {
+                      value: majorWithSpec.specialization,
+                      label: majorWithSpec.specialization.name,
+                    }
+                  : null
+              }
+              isDisabled={specsLoading[majorWithSpec.major.id]}
+              isLoading={specsLoading[majorWithSpec.major.id]}
+              onChange={async (data) => {
+                const updatedSpec = data?.value || null;
+                setResultsLoading((prev) => ({ ...prev, [majorWithSpec.major.id]: true }));
+                dispatch(setRequirements({ majorId: majorWithSpec.major.id, requirements: [] })); // set to empty immediately because otherwise it's out of date
+                dispatch(
+                  setSpecialization({
+                    majorId: majorWithSpec.major.id,
+                    specialization: updatedSpec,
+                  }),
+                );
+                await fetchRequirements(majorWithSpec.major.id, updatedSpec?.id || null);
+                const updatedMajors = selectedMajors.map((m) =>
+                  m.major.id === majorWithSpec.major.id ? { ...m, specialization: updatedSpec } : m,
+                );
+                saveMajors(updatedMajors);
+              }}
+              className="ppc-combobox"
+              classNamePrefix="ppc-combobox"
+              placeholder="Select a specialization..."
+              theme={(t) => comboboxTheme(t, isDark)}
+              onFocus={async () => {}}
+            />
+          )}
+          {resultsLoading[majorWithSpec.major.id] ? (
+            loadingIcon
+          ) : (
+            <ProgramRequirementsList requirements={majorWithSpec.requirements} />
+          )}
```

Comment:
I know this is a 'prettier' format, but having this across 5 lines makes this hard to read. After this is in a separate component, we should be able to fit it within its line length limit, hopefully.